### DISCUSSION TITLE
Error deploying a NEXT app with antd

### DISCUSSION THREAD

**Discussion by Jossdz** [vercel/next.js#14474] [Help] [answered]
![image](https://user-images.githubusercontent.com/23535629/85344536-9465c800-b4b5-11ea-8d26-f735f1a48d2d.png)

18:20:44.183  
Retrieving list of deployment files...
18:20:45.313  
Downloading 122 deployment files...
18:20:48.366  
Installing build runtime...
18:20:48.796  
Build runtime installed: 429.881ms
18:20:49.209  
Looking up build cache...
18:20:49.242  
Build cache not found
18:20:49.661  
Installing dependencies...
18:20:49.874  
yarn install v1.22.4
18:20:49.942  
[1/4] Resolving packages...
18:20:50.305  
[2/4] Fetching packages...
18:21:02.454  
info fsevents@2.1.3: The platform "linux" is incompatible with this module.
18:21:02.454  
info "fsevents@2.1.3" is an optional dependency and failed compatibility check. Excluding it from installation.
18:21:02.458  
info fsevents@1.2.13: The platform "linux" is incompatible with this module.
18:21:02.458  
info "fsevents@1.2.13" is an optional dependency and failed compatibility check. Excluding it from installation.
18:21:02.461  
[3/4] Linking dependencies...
18:21:02.464  
warning "antd > rc-picker@1.6.4" has unmet peer dependency "dayjs@^1.8.18".
18:21:08.640  
[4/4] Building fresh packages...
18:21:13.073  
Done in 23.20s.
18:21:13.098  
Running "yarn run build"
18:21:13.352  
yarn run v1.22.4
18:21:13.374  
$ cross-env NODE_OPTIONS="-r esm" next build
18:21:15.934  
Warning: No build cache found. Please configure build caching for faster rebuilds. Read more: https://err.sh/next.js/no-cache
18:21:15.935  
Creating an optimized production build...
18:21:15.952  
Attention: Next.js now collects completely anonymous telemetry regarding usage.
18:21:15.952  
This information is used to shape Next.js' roadmap and prioritize features.
18:21:15.952  
You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
18:21:15.952  
https://nextjs.org/telemetry
18:21:17.532  
> Using external babel configuration
18:21:17.532  
> Location: "/vercel/6d90a8c9/.babelrc"
18:21:45.523  
Compiled successfully.
18:21:45.523  
Automatically optimizing pages...
18:21:46.007  
> Build error occurred
18:21:46.008  
/vercel/6d90a8c9/node_modules/rc-notification/es/useNotification.js:1
18:21:46.008  
Error [ERR_REQUIRE_ESM]: Must use import to load ES Module: /vercel/6d90a8c9/node_modules/@babel/runtime/helpers/esm/toConsumableArray.js
18:21:46.008  
require() of ES modules is not supported.
18:21:46.008  
require() of /vercel/6d90a8c9/node_modules/@babel/runtime/helpers/esm/toConsumableArray.js from /vercel/6d90a8c9/node_modules/rc-notification/es/useNotification.js is an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which defines all .js files in that package scope as ES modules.
18:21:46.008  
Instead rename toConsumableArray.js to end in .cjs, change the requiring code to use import(), or remove "type": "module" from /vercel/6d90a8c9/node_modules/@babel/runtime/helpers/esm/package.json.
18:21:46.008  
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1172:13)
18:21:46.008  
    at Object.<anonymous> (/vercel/6d90a8c9/node_modules/rc-notification/es/useNotification.js:1)
18:21:46.008  
    at Generator.next (<anonymous>) {
18:21:46.008  
  code: 'ERR_REQUIRE_ESM'
18:21:46.009  
}
18:21:46.092  
error Command failed with exit code 1.
18:21:46.092  
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
18:21:46.100  
Error: Command "yarn run build" exited with 1
18:21:47.416  
Done with "package.json"

I'm having this error and I think it's 'cause I'm using Antd and the component using this dependency is an alert, but I don't know how fix this, help 🙏.

**Comment by gurmeetsinghmalhotra** [comment, marked as answer]
https://github.com/vercel/next.js/tree/canary/examples/with-ant-design
https://github.com/vercel/next.js/tree/canary/examples/with-ant-design-pro-layout-less
https://github.com/vercel/next.js/tree/canary/examples/with-ant-design-mobile
https://github.com/vercel/next.js/tree/canary/examples/with-ant-design-less
